docs/content/guides/dagster-pipes/subprocess.mdx

```diff
@@ -0,0 +1,56 @@
+---
+title: "Dagster Pipes tutorial | Dagster Docs"
+description: "Learn how to use Dagster Pipes's built-in subprocess implementation to invoke a subprocess with the given command and environment"
+---
+
+# Dagster Pipes tutorial
+
+In this guide, we’ll show you how to use [Dagster Pipes](/guides/dagster-pipes) with Dagster’s built-in subprocess implementation to run a subprocess with the given command and environment. You can then send information such as structured metadata and logging back to Dagster from the subprocess, where it will be visible in the Dagster UI.
+
+To get there, you'll
+
+- [Create a Dagster asset that invokes a subprocess](/guides/dagster-pipes/subprocess/create-subprocess-asset)
+- [Modify existing code to work with Dagster Pipes to send information back to Dagster](/guides/dagster-pipes/subprocess/modify-external-code)
+- Learn about other usage of Dagster Pipes with other entities in the Dagster system in the [Reference](/guides/dagster-pipes/subprocess/reference) section
+
+<Note>
+
+This guide focuses on using an out-of-the-box `PipesSubprocessClient` resource. For further customization with the subprocess invocation, use <PyObject module="dagster_pipes" object="open_dagster_pipes"/> approach instead. Refer to [Customizing Dagster Pipes protocols](/guides/dagster-pipes/customizing-dagster-pipes-protocols) for more info.
+
+</Note>
+
+---
+
+## Prerequisites
+
+To use Dagster Pipes to run a subprocess , you’ll need:
+
+- To have Dagster (`dagster`) and the Dagster UI (`dagster-webserver`) installed. Refer to the [Installation guide](/getting-started/install) for more info.
+
+- To install the dagster-pipes library:
+
+```bash
+pipe install dagster-pipes
+```
+
+- **An existing Python script**. We’ll use the following Python script to demonstrate. This file will be invoked by the Dagster asset that you’ll create later in this tutorial.
+  1. Create a standalone file named `external_code.py`.
+  2. Then, copy and paste the following into the file:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/external_code_original.py lines=2-
```

Comment:
unfortunately this is a limitation of the mdx-format 😢  when i intended correctly, the formatter would format it incorrectly

Reply:
tried it one more time. actually not mdx-format's fault. it's that when there are two empty lines in the code block + the block has a indentation (code on the right side), the markdown render couldn't render it correctly (rendered result on the left)
<img width="2035" alt="image" src="https://github.com/dagster-io/dagster/assets/4531914/ed1ce995-c8dd-4c88-8782-37b484f82a10">

Reply:
I found a workaround, but then the mdx-format script bongled it. 🤦‍♀️ The trick was to add a few spaces before those lines in the mdx file (lines 32-33), but the script strips them out.

This behavior doesn't make sense because Markdown rendering depends specifically on spacing, but that's a discussion for later. I think the best option (unless it's a quick fix in that script) is to re-work some of the copy so that lists aren't needed. I can help out here, since this occurs in a bunch of places.

Reply:
done structuring! let me know how you think

Reply:
Since this section is still being wonky, what if we did something like this to take out the list?

````
To use Dagster Pipes to run a subprocess, you’ll need to have Dagster (`dagster`) and the Dagster UI (`dagster-webserver`) installed. Refer to the [Installation guide](/getting-started/install) for more info.

You'll also need **an existing Python script.** We’ll use the following Python script to demonstrate. This file will be invoked by the Dagster asset that you’ll create later in this tutorial.

Create a file named `external_code.py` and paste the following into it:

```python file=/guides/dagster/dagster_pipes/subprocess/external_code_original.py lines=2-
import pandas as pd


def main():
    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
    total_orders = len(orders_df)
    print(f"processing total {total_orders} orders")


if __name__ == "__main__":
    main()
```
````

Reply:
oops - missed this one. fixed.